src/ruby/pb/test/client.rb

```diff
@@ -216,10 +233,29 @@ def each_item
   end
 end
 
+# Intended to be used to wrap a call_op, and to adjust
+# the write flag of the call_op in between messages yielded to it.
+class WriteFlagSettingStreamingInputEnumerable
+  attr_accessor :call_op
+
+  def initialize(requests_and_write_flags)
+    @requests_and_write_flags = requests_and_write_flags
+  end
+
+  def each
+    @requests_and_write_flags.each do |request_and_flag|
+      @call_op.write_flag = request_and_flag[:write_flag] if
+        request_and_flag[:write_flag]
```

Comment:
I think this won't do exactly what you want. It will never set the write flag to 0. Instead, I think you should set `@call_op.write_flag` unconditionally, and pass `0` for cases where no write flag should be set.